docs/standards/react.md

```diff
@@ -0,0 +1,170 @@
+  const [search, setSearch] = useState("");
+  const products = api.products.list.useQuery();
+  const filtered = products.filter(p => p.name.includes(search));
+  return { products: filtered, search, setSearch };
+};
+
+export const ProductListView = ({ products, search, setSearch }: UseProductListProps) => {
+  const [isFilterOpen, setIsFilterOpen] = useState(false); // ✅ UI state only
+  return <div><input value={search} onChange={e => setSearch(e.target.value)} /></div>;
+};
+
+```
+
+**Rule of thumb:** If it affects **what** data is shown → Hook. If it affects **how** it's shown → View.
+
+**Note:** Trivial formatting (`name.toUpperCase()`, `price.toFixed(2)`) is fine in View. Complex filtering/sorting/calculations with testable business rules must go in Hook for unit testability.
+
+If your view component contains a lot of valid local state, consider moving it to a separate hook and passing it in via the component hook. Including local state in the view component is useful when it is simple, but when it becomes complicated it should be moved out of the component for readability. 
+
+## Folder Structure (Next.js)
+
+Co-locate page components with routes, keep page-specific components in `_components`:
+
+```
+app/
+  home/
+    page.tsx                    # Route entry point
+    _components/
+      home.controller.tsx       # Orchestration
+      home.hook.ts              # Single hook when page isn't complex
+      home.hook.spec.ts         # Specs co-located
+      home.view.tsx
+      home.md                   # Docs co-located
+      other-component.tsx       # Simple specific to the home page
+      data-table/               # Complex component gets subdirectory
+        data-table.tsx
+        data-table.hook.ts
+        data-table.view.tsx
+        table-row.view.tsx      # Sub-views
+    _hooks/                     # Multiple hooks for complex pages
+      use-home-data.ts
+      use-home-actions.ts
+    _lib/                       # Page-specific helpers
+      helpers.ts
+
+```
+
+**Rules:**
+
+- `_components/` for page-specific components (**not reusable across pages**)
+- Shared/reusable components go in root `components/` directory
+- `_hooks/` for complex pages requiring multiple hooks
+- `_lib/` for page-specific helpers/utilities
+- Break complex pages into self-contained sections when possible
+- Spec files (`.spec.ts`, `.spec.tsx`) co-located next to implementation files
+- Documentation files (`.md`) co-located with features/components
+- `_` prefix keeps folders private to Next.js routing
+
+## Folder Structure (Vanilla React)
+
+Same controller-view-hook pattern, just under `features/` instead of `app/route/_components/`:
+
+```
+src/
+  features/
+    home/
+      home.tsx                  # Controller
+      home.hook.ts              # Single hook for simple features
+      home.hook.spec.ts
+      home.view.tsx
+      home.md
+      components/               # Feature-specific components
+        data-table/
```

Comment:
I know it's hard to come up with good examples (been there), but I wonder if data-table is a good name here, it's generic, makes me wonder if it should be extracted, how about `HomeTable`? or just `Table` (the folder provides context).

I'm also not a fan of the `components` folder, it will be hard to follow a domain structure if there exceptions.